caas/kubernetes/provider/metadata.go

```diff
@@ -81,83 +98,88 @@ func (k *kubernetesClient) GetClusterMetadata(storageClass string) (*caas.Cluste
 		if err != nil && !k8serrors.IsNotFound(err) {
 			return nil, errors.Trace(err)
 		}
-		if err == nil {
-			result.NominatedStorageClass = &caas.StorageProvisioner{
-				Name:        sc.Name,
-				Provisioner: sc.Provisioner,
-				Parameters:  sc.Parameters,
-			}
-			if sc.ReclaimPolicy != nil {
-				result.NominatedStorageClass.ReclaimPolicy = string(*sc.ReclaimPolicy)
-			}
+		if err == nil && sc != nil {
```

Comment:
if err == nil then sc will always be != nil right?

Reply:
right!